Refactor for scraped

ondenman opened the pull request:
Part of issue: https://github.com/everypolitician/everypolitician-data/issues/25302

This is a step towards importing the new term. The next step will be to change the term id from 2014 to 2017.

Note: I have moved the scraper to Morph everypolitician-scrapers

## [Scraper Change checklist](https://github.com/everypolitician/everypolitician/wiki/Scraper-Change-checklist)
* [x] 1. scraper is on Morph.io under the "everypolitician-scrapers" group?
* [x] 2. scraper's GitHub "Website" link points at morph.io page?
* [x] 3. scraper is set to auto-run?
* [x] 4. scraper is archiving?
* [x] 5. legislature has a scraper webhook set?

ondenman committed:
Extract MembersPage

ondenman committed:
Extract MemberPage

ondenman committed:
Move hardcoded data to scraper

tmtmtmtm commented on scraper.rb:
```diff
@@ -3,43 +3,32 @@
+    data = (scrape mem_url => MemberPage).to_h.merge(district: 'Alderney',
+                                                     party:    'Independent',
+                                                     term:     term)
+    data[:image] = URI.join(@BASE, URI.escape(data[:image])).to_s unless data[:image].to_s.empty?
```
Making the image absolute should happen in the class, not the consumer.

tmtmtmtm commented on lib/members_page.rb:
```diff
@@ -0,0 +1,9 @@
+require 'scraped'
+
+class MembersPage < Scraped::HTML
+  field :members do
```
These aren't really `members`: these are just the URLs. This should either return more information on each one, or be renamed as just `member_urls`.

ondenman committed:
Build image URL in MemberPage

ondenman committed:
Rename :members to :member_urls

This field returns urls.

tmtmtmtm commented on lib/member_page.rb:
```diff
@@ -11,7 +11,10 @@ class MemberPage < Scraped::HTML
+    base = 'http://www.alderney.gov.gg'
+    image = noko.css('div.limage img/@src').text
+    return image if image.empty?
+    URI.join(base, URI.escape(image)).to_s
```
Don't we have a Decorator to do this…?

ondenman committed:
Add decorator to make absolute urls from <base> url

The member pages provide their own base url within the `<base>` tag.

The standard Scraped decorator doesn't know how to handle this and so
we are rolling our own for this particular scraper.

ondenman committed:
Use AbsoluteURLsFromBase decorator in MemberPage

ondenman committed:
fixup! Build image URL in MemberPage

ondenman committed:
fixup! Extract MembersPage

ondenman commented:
Closing as the changes are now redundant.
